activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -839,7 +839,15 @@ def connection_pool_list
       end
       alias :connection_pools :connection_pool_list
 
-      def establish_connection(spec)
+      def establish_connection(spec_or_config, name: "primary")
+        if spec_or_config.is_a?(ConnectionSpecification)
+          spec = spec_or_config
+        else
+          resolver = ConnectionAdapters::ConnectionSpecification::Resolver.new(ActiveRecord::Base.configurations)
```

Comment:
Could shorten this a bit by leaving `ConnectionAdapters` and `ActiveRecord` implicit and rely on relative constant lookups 😁